core/jazz_slack-channel/index.js

```diff
@@ -0,0 +1,504 @@
+    })
+    .catch(error => {
+      logger.error(error);
+      if (error.result === "inputError") {
+        return cb(JSON.stringify(errorHandler.throwInputValidationError(error.message)));
+      } else if (error.result === "unauthorized") {
+        return cb(JSON.stringify(errorHandler.throwUnauthorizedError(error.message)));
+      } else {
+        return cb(JSON.stringify(errorHandler.throwInternalServerError(error.message)));
+      }
+    });
+};
+
+function genericInputValidation(event, config) {
+  logger.debug("Inside genericInputValidation");
+  return new Promise((resolve, reject) => {
+    if (!event || !event.body || !event.method) {
+      reject({
+        result: "inputError",
+        message: "Service inputs not defined!"
+      });
+    }
+
+    if (!event.body.channel_name) {
+      reject({
+        result: "inputError",
+        message: "channel_name is required!."
+      });
+    } else {
+      let name = event.body.channel_name;
+      if (name.length > config.slack_channel_name_length) {
+        reject({
+          result: "inputError",
+          message: "channel name should not exceed more than 21 characters"
+        });
+      } else if (/[^a-zA-Z0-9\-\_]/.test(name)) {
+        reject({
+          result: "inputError",
+          message: "channel name should accept letters, numbers, hyphens, and underscores"
+        });
+      }
+    }
+
+    if (!event.body.users || event.body.users.length < 1) {
+      reject({
+        result: "inputError",
+        message: "users.email_id is required!."
+      });
+    } else {
+      let users = event.body.users;
+      for (let i in users) {
+        let emailAddress = users[i].email_id;
+        if (!validateEmailId(emailAddress)) {
+          let msg = 'Not a valid email id!';
+          if (emailAddress) {
+            msg = emailAddress + " not a valid email id!.";
+          }
+          reject({
+            result: "inputError",
+            message: msg
+          });
+        } else if (emailAddress.toLowerCase() === (config.service_account_emailId).toLowerCase()) {
+          isServAccRequested = true;
+        }
+      }
+    }
+
+    if (event.method && event.method !== "POST") {
+      reject({
+        result: "inputError",
+        message: "Unsupported method/request."
+      });
+    }
+
+    if (!event.principalId) {
+      reject({
+        result: "unauthorized",
+        message: "Unauthorized."
+      });
+    }
+
+    resolve();
+  });
+}
+
+function identifyMembers(members, type, value) {
+  let info = null,
+  out = members.find(each => {
+    let memberEmailId = (each.profile.email) ? each.profile.email : null;
+    if (memberEmailId && type === 'id' && each.id === value) {
+      return each;
+    } else if (memberEmailId && type === 'email' && memberEmailId.toLowerCase() === value.toLowerCase()) {
+      return each;
+    }
+  });
+
+  info = formatData(out);
+  return info;
+}
+
+function isMemberAlreadyExists(id, list) {
```

Comment:
rename function name  to "doesMemberAlreadyExist"

Reply:
lines 150 - 157 is equivalent to:
`return list.find(each => each.id === id) != undefined;`

Reply:
Updated as per last comment